driver-core/src/main/java/com/datastax/driver/core/policies/FallthroughRetryPolicy.java

```diff
@@ -86,4 +86,17 @@ public RetryDecision onWriteTimeout(Statement statement, ConsistencyLevel cl, Wr
     public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int requiredReplica, int aliveReplica, int nbRetry) {
         return RetryDecision.rethrow();
     }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onUnexpectedError(Statement statement, ConsistencyLevel cl, int nbRetry, boolean mightHaveBeenApplied) {
+        return RetryDecision.tryNextHost(cl);
```

Comment:
tryNextHost(cl) is consistent with current behavior, but rethrow() might be more appropriate here?